### PR TITLE
Customizing sidebar items from templates

### PR DESCRIPTION
**Pull request by alanlomeli**
Here we'll completely get rid off the sidebar generation from a script to use the new FSDocs feature that allows menu-items and headers customization from a template.

### CODE REVIEW

File: paket.lock [hunk 1]
```diff
@@ -172,115 +172,96 @@ GITHUB
...
-      Microsoft.Build.Tasks.Core
-      Microsoft.Build.Utilities.Core
-      Microsoft.CodeAnalysis.CSharp (>= 3.6 < 4.0)
-      Microsoft.NET.Test.Sdk
-      Newtonsoft.Json
-      NUnit (>= 3.12 < 4.0)
-      NUnit3TestAdapter
-      Suave
-      System.Memory
+    src/Compiler/AbstractIL/il.fs (f8b56cb6db3c90348810263fd7a5d37dc5719e92)
```
Comment (nojaf) [inline paket.lock:175]: You are reverting the compiler to a different state here. A local build (`dotnet fake build`) could never have worked.

File: docs/_template.html [hunk 2]
```diff
@@ -33,7 +33,7 @@
       <div id="sidebar-logo" class="d-flex justify-content-center p-5">
       <a href="./index.html" ><img id="fsdocs-logo" src="{{root}}/{{fsdocs-logo-src}}" /></a>
 </div>
-      <div class="list-unstyled ps-0 text-white fs-6">
+      <div id=sidebar-generation" class="list-unstyled ps-0 text-white fs-6">
```
Comment (nojaf) [inline docs/_template.html:36]: Nitpick, but as nothing is no longer generated you should consider to change the name of this identifier.

File: docs/_menu_template.html [hunk 3]
```diff
@@ -0,0 +1,16 @@
+            <div id="{{fsdocs-menu-header-id}}-wrapper" class="p-2 list-wrapper menu-open">
+            <li class="docs-menu">
+                <button class="btn text-white d-flex p-1 justify-content-between" data-bs-toggle="collapse"
```
Comment (nojaf) [inline docs/_menu_template.html:3]: Add "align-items-center" as well to align the text and the icon

File: docs/_menu_template.html [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+                        data-bs-target="#{{fsdocs-menu-header-id}}" aria-expanded="true">
+                        {{fsdocs-menu-header-content}}
+            <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" stroke="currentColor" stroke-width="2" fill="currentColor" class="bi bi-chevron-down mx-1" viewBox="0 0 16 16">
```
Comment (nojaf) [inline docs/_menu_template.html:6]: I'd go for an `<i>` tag here instead, something like:
```html
<i class="bi bi-chevron-down mx-2" style="-webkit-text-stroke: 1px;"></i>
```

File: docs/_menu_template.html [hunk 5]
```diff
@@ -0,0 +1,16 @@
+            <div id="{{fsdocs-menu-header-id}}-wrapper" class="p-2 list-wrapper menu-open">
+            <li class="docs-menu">
```
Comment (nojaf) [inline docs/_menu_template.html:2]: The background doesn't change anymore when the menu opens.

**Review by nojaf — CHANGES_REQUESTED** [review]
This PR proves our idea is working but this whole thing is in a broken state.
Please address these problems. If you are aware that some things are still in progress, please create a mini to-do list in the PR description.

File: docs/_menu_template.html (hunk 4 above)
Comment (yisusalanpng) [reply docs/_menu_template.html:6]: sure, do note that I added a vector because it's font counter part didn't allow me to resize it the way I liked. I'll look into it tho.

**Comment by yisusalanpng** [comment]
TODO:
Some stylings for divs with class 'menu-open' are not being applied. Therefore the background color for the menu and animations are not working yet.

**Comment by nojaf** [comment]
Regarding `fsdocs-default.css`, you should keep it in the `.style` folder and update the import statement to `@import "./fsdocs-default"` (without the file extension) in `fsdocs-custom.sass`.
This will included the entire CSS file in the sass output, instead of generating an `@import url(...)`.

**Comment by nojaf** [comment]
I actually want the main menu to be open by default but sub menus collapsed.

image

In a later PR we can then try to open the correct menu by default. (Not for this PR!).

File: docs/_template.html [hunk 6]
```diff
@@ -33,8 +33,9 @@
       <div id="sidebar-logo" class="d-flex justify-content-center p-5">
       <a href="./index.html" ><img id="fsdocs-logo" src="{{root}}/{{fsdocs-logo-src}}" /></a>
 </div>
-      <div id="sidebar-generation" class="list-unstyled ps-0 text-white fs-6">
+      <div id=sidebar-content" class="list-unstyled ps-0 text-white fs-6">
```
Comment (nojaf) [inline docs/_template.html:36]: This was a tricky one, there is no opening `"` for the `id` attribute here.
This leads to the id not being set and introduces the menu colour change bug.

File: docs/_menu_template.html [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+                <button class="btn text-white d-flex p-1 justify-content-between align-items-center" data-bs-toggle="collapse"
+                        data-bs-target="#{{fsdocs-menu-header-id}}" aria-expanded="true">
+                        {{fsdocs-menu-header-content}}
+                        <i class="bi bi-chevron-down mx-2" style="-webkit-text-stroke: 1px;"></i>
+                </button>
+                <div id="{{fsdocs-menu-header-id}}" class="ms-4 collapse show">
+                    <ul class="list-unstyled fw-normal pb-1">
```
Comment (nojaf) [inline docs/_menu_template.html:9]: Maybe add a `mb-0` here as well to remove the margin bottom.

File: docs/_template.html (hunk 6 above)
Comment (yisusalanpng) [reply docs/_template.html:36]: that was fun, thanks!

File: docs/.style/fsdocs-custom.sass [hunk 8]
```diff
@@ -32,58 +33,58 @@ $headings-font-family: 'Reem Kufi', sans-serif
...
-      a
-        color: $black
-        text-decoration: none
-        text-decoration-style: none
+  box-shadow: 0 4px 10px $drop-shadow
+  h1, h2, h3, h4, h5, h6
+    a
+      color: $black
+      text-decoration: none
+      text-decoration-style: none
```
Comment (nojaf) [inline docs/.style/fsdocs-custom.sass:49]: Invalid CSS property: https://www.w3schools.com/cssref/css3_pr_text-decoration-style.asp

File: docs/.style/homepage.sass [hunk 9]
```diff
@@ -85,7 +85,7 @@ header
     transform: translate(-50%, -50%)
 
   #cta-buttons
-    z-index:1
+    z-index: 1
```
Comment (nojaf) [inline docs/.style/homepage.sass:88]: This again doesn't belong to this PR. Don't touch files that are unrelated to the current goal.

**Comment by nojaf** [comment]
https://www.nuget.org/packages/FSharp.Formatting/16.0.1 was released, @alanlomeli please update to this version and remove all the `paket` shannigans.

Revert `paket.dependencies`, `paket.lock` and update the `FAKE` target accordingly.
Please this everything out locally first 😊.

**Review by nojaf — APPROVED** [review]
Looks good!
